### PR TITLE
feat: creates payment slip processor

### PR DESCRIPTION
**Pull request by jaugustorufino**
## Development
* Creates a Payment Slip Processor using TDD

### Features
- Creates `Invoice` entity
- Creates `PaymentSlip` entity
- Creates `PaymentProcessor` entity and implements logic of the processing of some payment slips
- Creates `Payment` entity
- Creates `PaymentType` enum

### CODE REVIEW

File: payment-processor/src/main/java/Invoice.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class Invoice {
```
Comment (pedrohenrique-ql) [inline payment-processor/src/main/java/Invoice.java]: Adopting a single language for the code is essential for consistency and readability. What do you think of changing class names to Portuguese or variable and method names to English to standardize the language used in the code?

File: payment-processor/src/main/java/Invoice.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    private String nomeCliente;
+    private final double valor;
+    private double valorAtual;
+    private String data;
+    private InvoiceStatus status;
+    private List<Payment> pagamentos;
+
+    public Invoice(String nome, double valor, String data) {
+        this.nomeCliente = nome;
+        this.valor = valor;
```
Comment (pedrohenrique-ql) [inline payment-processor/src/main/java/Invoice.java]: What do you think about rename the variable `valor` to `valorTotal` to emphasize its semantics?

File: payment-processor/src/test/java/InvoiceTest.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.*;
+
+public class InvoiceTest {
+
+    @Test
+    public void testFatura() {
```
Comment (pedrohenrique-ql) [inline payment-processor/src/test/java/InvoiceTest.java]: The name of the test could be `testCriacaoFatura`, what do you think?

File: payment-processor/src/test/java/PaymentSlipTest.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class PaymentSlipTest {
+
+    @Test
+    public void testBoleto() {
```
Comment (pedrohenrique-ql) [inline payment-processor/src/test/java/PaymentSlipTest.java]: Similarly, I believe it's better to be renamed to `testCriacaoBoleto`.

File: payment-processor/src/test/java/PaymentTest.java [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class PaymentTest {
+
+    @Test
+    public void testPagamento() {
```
Comment (pedrohenrique-ql) [inline payment-processor/src/test/java/PaymentTest.java]: Likewise, `testCriacaoPagamento` better indicates the purpose of the test.

**Review by pedrohenrique-ql — CHANGES_REQUESTED** [review]
Nice job, just a few comments.

File: payment-processor/src/main/java/Invoice.java (hunk 1 above)
Comment (jaugustorufino) [reply payment-processor/src/main/java/Invoice.java]: I'm gonna change the class names to Portuguese :+1:

**Review by pedrohenrique-ql — APPROVED** [review]
Nice!

**Pull request merged**